### PR TITLE
Use absolute paths in requirements files

### LINKED ISSUE #813: Relative paths in requirements files don't work
Requirements in pyproject.toml files can be in any form accepted by `pip install`, including paths to local sdists, wheels, or directories. Paths are interpreted relative to the location of the pyproject.toml file.

However, when these paths are copied into a requirements.txt file for Flatpak or Chaquopy, they'll be interpreted relative to the location of the requirements file, which doesn't currently work.

Possible solutions:
* Generate the requirements file in the same directory as pyproject.toml. Not ideal, because it would be a platform-specific file generated outside of the platform directory
* Convert all relative paths in the requirements file to absolute paths. This is easy, but would prevent the project directory from being relocated.
* Convert all relative paths in the requirements file to be relative to the file's location. I think this is the best solution, and there's a `os.path.relpath` function that should make it just as easy.

### PR DESCRIPTION
**Pull request by mhsmith**
Fixes #813. I had to use absolute paths in the end, because the Flatpak build moves the requirements file a second time before using it.

In order to get this working with Flatpak, the following additional changes were needed:
* https://github.com/beeware/briefcase-linux-flatpak-template/pull/4, so local requirement paths are visible to the Flatpak build process.
* Fixed Flatpak's override of the support package install method, which was broken by #805. This wasn't caught by the tests because they tested the behavior of the overridden low-level method, but not whether it was actually called. So I changed the tests to call a higher-level method.

## PR Checklist:

- [x] All new features have been tested
- [x] All new features have been documented
- [x] I have read the **CONTRIBUTING.md** file
- [x] I will abide by the code of conduct

### CODE REVIEW

**Review by freakboy3742 — CHANGES_REQUESTED** [review]
Good catch on the flatpak support package refactoring problem. Add that one to the list of problems that would have been caught if we actually ran apps as part of CI (See #699). 

One minor tweak regarding `os.path` vs `pathlib` which I've made on your behalf; and one larger problem with other valid inputs to requirements. I've reworked and extended the test cases to highlight the problem.

**Comment by mhsmith** [comment]
Thanks, I'll look over this tomorrow.

File: src/briefcase/commands/create.py [hunk 1]
```diff
@@ -402,10 +402,18 @@ def _write_requirements_file(self, app: BaseConfig, requirements_path):
...
+            separators.append(os.altsep)
+
         with self.input.wait_bar("Writing requirements file..."):
             with (requirements_path).open("w", encoding="utf-8") as f:
                 if app.requires:
                     for requirement in app.requires:
+                        # Update paths to be absolute, because flatpak moves the requirements
+                        # file to a different place before using it.
+                        if any(sep in requirement for sep in separators):
+                            requirement = os.path.abspath(self.base_path / requirement)
```
[the remark this answers is not in the document]
Comment (mhsmith) [reply src/briefcase/commands/create.py]: I can't think of any benefit of resolving symbolic links here, but I can think of a potential problem. Consider the following sequence of events:
* The user adds a requirement path with a symlink, then runs `briefcase update -d`. A requirements.txt file is generated containing the symlink target path.
* The user changes the target of the symlink.
* The user runs `briefcase build`. The existing requirements.txt file is used, with the old symlink target.

The user's mental model is that their requirements list is passed directly to pip: the requirements file is just an implementation detail. So it's reasonable for them to assume that symlinks will be resolved by pip itself.

[the remark this answers is not in the document]
Comment (freakboy3742) [reply src/briefcase/commands/create.py]: That's a reasonable point... I defaulted to "resolving references is a good thing"; but thinking about it some more, I think you're right that we should preserve the symlink, and let the last tool to *use* the link do the resolution.

**Review by freakboy3742 — APPROVED** [review]
👍 Nice work!

**Pull request merged**